src/agents/runner.ts

```
@@ -0,0 +1,283 @@
+
+/**
+ * Extract the first balanced JSON object from content.
+ * Uses a brace-depth scanner to handle nested objects correctly.
+ */
+function extractFirstJsonObject(content: string): string | null {
+  const startIdx = content.indexOf("{");
+  if (startIdx === -1) return null;
+
+  let depth = 0;
+  let inString = false;
+  let escape = false;
```

Biome (2.1.2)

[error] Do not shadow the global "escape" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)